gnomad/utils/filtering.py

```diff
@@ -460,12 +469,51 @@ def filter_to_gencode_cds(
         "This Gencode CDS interval filter does not filter by transcript! Please see the"
         " documentation for more details to confirm it's being used as intended."
     )
-    filter_expr = hl.is_defined(gencode_ht[t.locus])
+    if genes:
+        genes_upper = (
+            [genes.upper()] if isinstance(genes, str) else [g.upper() for g in genes]
+        )
+        if by_gene_symbol:
+            gencode_ht = gencode_ht.filter(
+                hl.literal(genes_upper).contains(gencode_ht.gene_name)
+            )
+        else:
+            gencode_ht = gencode_ht.filter(
+                hl.literal(genes_upper).contains(gencode_ht.gene_id)
+            )
 
-    if isinstance(t, hl.MatrixTable):
-        t = t.filter_rows(filter_expr)
+    if padding_bp:
+        gencode_ht = gencode_ht.annotate(
+            padded_interval=hl.locus_interval(
+                gencode_ht.interval.start.contig,
+                gencode_ht.interval.start.position - padding_bp,
+                gencode_ht.interval.end.position + padding_bp,
+                includes_start=gencode_ht.interval.includes_start,
+                includes_end=gencode_ht.interval.includes_end,
+                reference_genome=gencode_ht.interval.start.dtype.reference_genome,
+            )
+        )
+
+    # Collect intervals if there are more than `max_intervals` to avoid memory issues
+    if gencode_ht.count() < max_intervals:
```

Comment:
num_intervals = gencode_ht.count()
```suggestion
    if num_intervals <= max_collect_intervals:
```